### PR TITLE
PaymentResponse.details attribute toJSON()

### LINKED ISSUE #7319: PaymentResponse missing tests
- [x] .toJSON() 
- [x] details attribute (just need to test if its an object, not IDL conversion... will do that in Basic Cards tests)
- [x] shippingAddress attribute (just type check).

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #7319

### CODE REVIEW

**Comment by marcoscaceres** [comment]
I think this is the last batch of tests 🤞 🎉  

This is probably sufficient tests for a 1.0. 

Subjectively*, we now  have 100% coverage of the spec now. 
 
* me going over every single line with a printed spec using a pen to check that each assertion is tested.

**Comment by marcoscaceres** [comment]
Actually, no... I celebrated too soon ... gotta test https://github.com/w3c/web-platform-tests/issues/7241 still ... so close 🤤

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-09-18 08:57:17
Finished: 2017-09-18 09:10:10


<h2>Failing Jobs</h2>
<ul>

<li>build_css</li>

<li>chrome:unstable</li>

<li>safari:10.0</li>

<li>MicrosoftEdge:14.14393</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18605)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/276758428)

**Review by rsolomakhin — APPROVED** [review]

File: payment-request/PaymentAddress/attributes-and-toJSON-method-manual.https.html [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      const msg = `Expected JSON ${prop} to be ${expectedValue}`;
+      assert_equals(actualValue, expectedValue, msg);
+    }
+  }, button.textContent.trim());
+  done();
+}
+</script>
+<h2>PaymentAddress inteface</h2>
+<p>
+  When prompted, please enter enter addresses as follows...
```
Comment (edenchuang) [inline payment-request/PaymentAddress/attributes-and-toJSON-method-manual.https.html]: two "enter"s.

**Review by edenchuang — APPROVED** [review]

File: payment-request/PaymentAddress/attributes-and-toJSON-method-manual.https.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+<meta charset="utf8">
+<link rel="help" href="https://www.w3.org/TR/payment-request/#paymentaddress-interface">
+<title>
+  PaymentResponse.prototype.shippingAddress
+</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../payment-response/helpers.js"></script>
+<script>
+const options = { requestShipping: true };
```
Comment (edenchuang) [inline payment-request/PaymentAddress/attributes-and-toJSON-method-manual.https.html]: setup({ explicit_done: true, explicit_timeout: true });

**Review by edenchuang — APPROVED** [review]

File: payment-request/PaymentAddress/attributes-and-toJSON-method-manual.https.html (hunk 2 above)
Comment (marcoscaceres) [reply payment-request/PaymentAddress/attributes-and-toJSON-method-manual.https.html]: It's added by helpers.js in this case.

File: payment-request/payment-response/helpers.js [hunk 3]
```diff
@@ -113,6 +113,13 @@ async function runManualTest(button, options, expected = {}, id = undefined) {
         `Expected response ${attribute} attribute to be ${value}`
       );
     }
+    assert_idl_attribute(response, "details");
+    assert_equals(typeof response.details, "object", "Expected an object");
+    try {
```
Comment (domenic) [inline payment-request/payment-response/helpers.js]: This will produce a less-helpful error message. Instead, just call toJSON(), and let it throw.

Comment (marcoscaceres) [reply payment-request/payment-response/helpers.js]: good point. Will send as a new PR.

**Pull request merged**